snorkel/labeling/apply/core.py

```diff
@@ -33,9 +33,11 @@ def __init__(self, lfs: List[LabelingFunction]) -> None:
         self._lfs = lfs
 
     def _matrix_from_row_data(self, labels: List[RowData]) -> np.ndarray:
-        row, col, data = zip(*chain.from_iterable(labels))
         L = np.zeros((len(labels), len(self._lfs)), dtype=int) - 1
-        L[row, col] = data
+        # NB: this check will short-circuit, so ok for large L
+        if any(map(len, labels)) > 0:
```

Comment:
any will return True when it finds a non-zero entry, so you don't need the '> 0' at the end.

Reply:
hmmmm well that's what was happening anyways because i had a pretty trick bug (which ends up having no impact). i meant to do 

```
any(map(lambda x: len(x) > 0, labels))
```